pkgs/applications/editors/rstudio/server.nix

```diff
@@ -0,0 +1,145 @@
+{ lib, mkDerivation, fetchurl, fetchpatch, fetchFromGitHub, makeDesktopItem, cmake, boost, zlib
+, openssl, R, qtbase, qtxmlpatterns, qtsensors, qtwebengine, qtwebchannel
+, libuuid, hunspellDicts, unzip, ant, jdk, gnumake, makeWrapper, pandoc
+, llvmPackages, readline6, pam
+}:
+
+with lib;
+let
+  verMajor = "1";
+  verMinor = "2";
+  verPatch = "5042";
+  version = "${verMajor}.${verMinor}.${verPatch}";
```

Comment:
This can be done functionally with lib.versions.majorMinor

Reply:
thanks, manually wrote in the full version and programmatically accessed the major minor and patches.